sdk/core/core-amqp/src/retry.ts

```diff
@@ -110,6 +120,22 @@ async function checkNetworkConnection(host: string): Promise<boolean> {
   }
 }
 
+async function executeOperation<T>(config: RetryConfig<T>) {
+  let isOperationDone = false;
+  setTimeout(() => {
```

Comment:
I think you want to clear this timeout when the request has succeeded or failed. You will probably need to handle errors because ISTM this code will throw an error if the operation throws too (isOperationDone never gets set to true).

Reply:
Yes, that makes sense! I've just updated this.